release_creation/snapshots.py

```diff
@@ -0,0 +1,36 @@
+from sqlite3 import adapters
+import subprocess
+import os
+import shutil
+from typing import Optional
+from semantic_version import Version
+
+_BASE_DOWNLOAD_CMD = """pip3 download --dest ./tmp """
+_OUTPUT_ARCHIVE_FILE_BASE = "dbt-core-all-adapters-snapshot"
+_FILE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+# def _generate_package_req(version: dict):
+#     suffix = version['suffix'] if 'suffix' in version else ''
+#     return f"{version['name']}{version['version_pinning']}{version['sem_ver']} {suffix} \\\n"
+
+def _generate_download_command(major_version:Optional[int], minor_version:Optional[int], is_pre:bool = False) -> str:
+    download_command = _BASE_DOWNLOAD_CMD
+    suffix = "latest"
+    if is_pre:
+        download_command += "--pre"
+        suffix = "pre"
+    download_command += f" -r {_FILE_DIR}/requirements/v{major_version}.{minor_version}.{suffix}.requirements.txt"
```

Comment:
Do we need additional cases to generate download_command here if `major_version` or `minor_version` are `None`?